src/recordlinker/linkage/models.py

```diff
@@ -47,3 +44,48 @@ class BlockingValue(Base):
     patient_id: orm.Mapped[int] = orm.mapped_column(ForeignKey("mpi_patient.id"))
     blockingkey_id: orm.Mapped[int] = orm.mapped_column(ForeignKey("mpi_blocking_key.id"))
     value: orm.Mapped[str] = orm.mapped_column(String(50), index=True)
+
+class Algorithm(Base):
+    __tablename__ = "algorithm"
+
+    id: orm.Mapped[int] = orm.mapped_column(primary_key=True)
+    is_default: orm.Mapped[bool] = orm.mapped_column(default=False)
+    label: orm.Mapped[str] = orm.mapped_column(String(255), unique=True)
+    description: orm.Mapped[str]
+
+def check_only_one_default(connection, target):
+    """
+    Check if there is already a default algorithm before inserting or updating.
+
+    Called before an insert or update operation on the
+    Algorithm table. If the `is_default` attribute of the target object is
+    set to True, it checks the database to ensure that no other algorithm
+    is marked as default. If another default algorithm exists, an exception
+    is raised to prevent the operation.
+
+    Parameters:
+    connection: The database connection being used for the operation.
+    target: The instance of the Algorithm class being inserted or updated.
+    
+    Raises:
+    Exception: If another algorithm is already marked as default.
+    """
+     
+    if target.is_default:
+        existing_default = connection.execute("SELECT COUNT(*) FROM algorithm WHERE is_default = TRUE").scalar()
+        if(existing_default > 0):
+            raise ValueError("There can only be one default algorithm.")
+
+event.listen(Algorithm, 'before_insert', check_only_one_default)
```

Comment:
Could we add some test cases for these situations to make sure the `check_only_one_default` works as expected.

Reply:
@ericbuckley Was looking at adding a test_models.py file as a place to put the test for this function but it looks like you created one in https://github.com/CDCgov/RecordLinker/pull/30

Reply:
test_models.py makes sense, we can rebase appropriately after one of us merges.